crates/compiler/can/src/suffixed.rs

```diff
@@ -813,33 +836,120 @@ fn unwrap_low_level_dbg<'a>(
+
+            // desugar to
+            // Task.await
+            //     (
+            //         #!0_expr : Task loc_type _
+            //         #!0_expr = loc_expr
+            //         #!0_expr
+            //     )
+            //     \loc_pat -> loc_cont
+            use roc_parse::ast::*;
+
+            // #!a0
+            let new_ident = next_unique_suffixed_ident();
+            let new_ident = match loc_pat.value {
+                Pattern::Underscore("#!stmt") => format!("{}_stmt", new_ident),
+                Pattern::Identifier { ident }
+                    if ident.starts_with('#') && ident.ends_with("_stmt") =>
+                {
+                    format!("{}_stmt", new_ident)
+                }
+                _ => format!("{}_expr", new_ident),
+            };
+            let new_ident = arena.alloc(new_ident);
+
+            // #!0_expr (pattern)
+            // #!0_expr : Task loc_type _
+            // #!0_expr = loc_expr
+            let value_def = ValueDef::AnnotatedBody {
+                ann_pattern: arena.alloc(Loc::at(
+                    loc_ann_pat.region,
+                    Pattern::Identifier {
+                        ident: if loc_ann_pat.value.equivalent(&loc_pat.value) {
+                            new_ident
+                        } else {
+                            // create another pattern to preserve inconsistency
+                            arena.alloc(next_unique_suffixed_ident())
+                        },
+                    },
+                )),
+                ann_type: arena.alloc(Loc::at(
+                    loc_type.region,
+                    TypeAnnotation::Apply(
+                        arena.alloc(""),
+                        arena.alloc("Task"),
+                        arena.alloc([
+                            *loc_type,
+                            Loc::at(loc_type.region, TypeAnnotation::Inferred),
+                        ]),
+                    ),
+                )),
+                comment: None,
+                body_pattern: arena.alloc(Loc::at(
+                    loc_pat.region,
+                    Pattern::Identifier { ident: new_ident },
+                )),
+                body_expr: loc_expr,
+            };
+
+            // #!0_expr (variable)
+            let new_var = arena.alloc(Loc::at(
+                region,
+                Expr::Var {
+                    module_name: "",
+                    ident: new_ident,
+                },
+            ));
+
+            // (
+            //     #!0_expr : Task loc_type _
+            //     #!0_expr = loc_expr
+            //     #!0_expr
+            // )
+            let mut defs = roc_parse::ast::Defs::default();
+            defs.push_value_def(value_def, region, &[], &[]);
+
+            arena.alloc(Loc::at(
+                Region::span_across(&loc_ann_pat.region, &loc_expr.region),
+                Defs(arena.alloc(defs), new_var),
+            ))
+        }
+        _ => {
```

Comment:
Is it clearer if this is `None`?

Reply:
Agree, will change

Reply:
fixed in https://github.com/roc-lang/roc/pull/6868/commits/3b50a8322a76c3e8f59d3cba555e2bc2754e260d